src/Host.cc

```diff
@@ -0,0 +1,101 @@
+// -*- lsst-c++ -*-
+
+/* 
+ * LSST Data Management System
+ * Copyright 2008-2014  AURA/LSST.
+ * 
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/** \file Host.cc
+  *
+  * \brief quickly return the host ip address
+  *
+  * \ingroup events
+  *
+  */
+#include <iomanip>
+#include <sstream>
+#include <stdexcept>
+#include <string.h>
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <netdb.h>
+
+#include "lsst/ctrl/events/Host.h"
+#include "lsst/pex/exceptions.h"
+
+namespace pexExceptions =lsst::pex::exceptions;
+
+using namespace std;
+
+namespace lsst {
+namespace ctrl {
+namespace events {
+
+/** \brief return the default Host object, which can access the host
+  *              and ip address.
+  * \return The EventSystem object
+  */
+Host& Host::getHost() {
+    if (thisHost == 0) {
+
+        // create the _IPAddr here, rather than
+        // reconstructing it every time we create an
+        // identificationId
+
+        char buf [255];
+        struct hostent *ent;
```

Comment:
This pre-declaration is C.  In C++ (and C99) you should declare them only as you use them.  And then you can make them `const`.